### ISSUE TITLE
Retorno sem código do IBGE

### ISSUE THREAD

**Issue by petry26** [PostmonAPI/postmon#154]
Gostaria de informar que a consulta do  CEP 89533000, retorna o código IBGE do estado, mas não da  cidade. Seria isso um bug da API?

**Comment by iurisilvio** [comment]
Sim, é um bug. O nome na lista do IBGE que a gente usa o nome é um pouco diferente (`Cristovão` sem acento no `ó`).

http://www.ibge.gov.br/home/geociencias/areaterritorial/area.php?nome=%&codigo=&submit.x=31&submit.y=2

Esse caso é possível de corrigir limpando os acentos antes de comparar, mas é difícil zerar as diferenças entre cidades, tem alguns casos um pouco mais difíceis de tratar.

**Comment by andersonscherrrer** [comment]
Complementando a questão o cep 42840-000, no postmon aparece o nome do município Abrantes (Camaçari), e na consulta ibge http://www.ibge.gov.br/home/geociencias/areaterritorial/area.php?nome=% ele aparece apenas como Camaçari.
Podemos fazer alguma coisa para informar o alias (um novo campo na tabela municípios, e caso o município exista no ibge, mas não seja encontrado no postmon, podemos tratar para verificar se tem o alias e mesma uf.

**Comment by andersonscherrrer** [comment]
Eu não manjo de python mas, seria algo assim:
 def insert_or_update_cidade(self, obj, **kwargs):
        update = {'$set': obj}
        chave = 'sigla_uf_nome_cidade'
        chave_alias = 'sigla_uf_alias_cidade'
        self._db.cidades.update($or[{chave: obj[chave]} , {chave_alias: obj[chave]}], update, upsert=True)

**Comment by iurisilvio** [comment]
@andersonscherrrer uma tabela dessas é simples de implementar, a dificuldade maior é manter essa relação dos aliases.

Essa é a solução do problema, porque qualquer tentativa de solução automática ainda vai ter algum caso específico que vai precisar cair nesse caso. Dá para melhorar o lookup automático para reduzir a necessidade de aliases, mas no fim alguns casos vão cair nessa tabela de aliases.

**Comment by iurisilvio** [comment]
Para facilitar o entendimento e pensar em soluções, fiz um dump das cidades de produção.

https://gist.github.com/iurisilvio/4115eee893d08be9e59691084abbd533

Dá para ver que uma solução simples talvez seja buscar pelo nome em parenteses quando existir. Deve resolver muitos casos, tem que só entender se não vai gerar falsos resultados.

Vou abrir issues para cada um dos problemas.

**Comment by iurisilvio** [comment]
Tratados os dois problemas principais. Ainda existem alguns poucos casos problemáticos.

A maioria dos que eu verifiquei, são problemas de grafia. Poderia tentar um match por distância das palavras.

```
BA;Jequiriçá
BA;Santa Terezinha
GO;Bom Jesus
MA;Governador Edson Lobão
MA;Pimentel (Pindaré Mirim)
MA;Pindaré Mirim
MA;Santo Amaro
MA;Senador La Roque
MG;Amparo da Serra
MG;Brasópolis
MG;Dias (Brasópolis)
MG;Dona Euzébia
MG;Luminosa (Brasópolis)
MG;Olhos D'Água
MG;Padre Felisberto (Amparo da Serra)
MG;Paraíso Garcia (Santa Rita do Ibitipoca)
MG;Passa Vinte
MG;Santa Rita do Ibitipoca
MG;Sem Peixe
MG;São Manoel do Guaiaçu (Dona Euzébia)
PA;Americano (Santa Isabel do Pará)
PA;Caraparu (Santa Isabel do Pará)
PA;Eldorado dos Carajás
PA;Santa Isabel do Pará
PE;Iguaraci
PE;Irajaí (Iguaraci)
PE;Jabitaca (Iguaraci)
PE;Lagoa do Itaenga
RJ;Parati
RJ;Parati Mirim (Parati)
RJ;Tarituba (Parati)
RN;Arez
RN;Boa Saúde
RN;Campo Grande
RN;Córrego de São Mateus (Boa Saúde)
RO;Boa Vista de Pacarana (Espigão do Oeste)
RO;Espigão do Oeste
SE;Graccho Cardoso
SP;Catucaba (São Luiz do Paraitinga)
SP;São Luiz do Paraitinga
TO;Apinajé (São Valério da Natividade)
TO;Couto de Magalhães
TO;São Valério da Natividade
```
